phpstan.neon

```diff
@@ -0,0 +1,25 @@
+parameters:
+    autoload_directories:
+        - %currentWorkingDirectory%/tests/Doctrine/Tests/Common/ClassLoaderTest
+    autoload_files:
+        - %currentWorkingDirectory%/tests/Doctrine/Tests/Common/Persistence/ObjectManagerDecoratorTest.php
+        - %currentWorkingDirectory%/tests/Doctrine/Tests/Common/Persistence/PersistentObjectTest.php
+        - %currentWorkingDirectory%/tests/Doctrine/Tests/Common/Persistence/Mapping/ClassMetadataFactoryTest.php
+        - %currentWorkingDirectory%/tests/Doctrine/Tests/Common/Persistence/Mapping/_files/annotation/TestClass.php
+    excludes_analyse:
+        - %currentWorkingDirectory%/lib/vendor/doctrine-build-common
+        - %currentWorkingDirectory%/tests/Doctrine/Tests/Common/Proxy/InvalidReturnTypeClass.php
+        - %currentWorkingDirectory%/tests/Doctrine/Tests/Common/Proxy/InvalidTypeHintClass.php
+        - %currentWorkingDirectory%/tests/Doctrine/Tests/Common/Persistence/Mapping/_files/TestEntity.php
+        - %currentWorkingDirectory%/tests/Doctrine/Tests/Common/Proxy/generated
+    ignoreErrors:
+        - '#Doctrine\\Common\\Reflection\\StaticReflection[a-zA-Z0-9_]+::__construct\(\) does not call parent constructor from Reflection[a-zA-Z0-9_]+#'
+        - '#\(void\) is used#'
```

Comment:
This (and `with return type void returns void but should not return anything`) both happen in ObjectManagerDecorator because a lot of methods have `@return void`  but:

1) Do not have `return;`, but `return something;`
2) And this `something` is also a method with `@return void`.

It's these places concretely:

```
 ------ ---------------------------------------------------------------------------------------------------------------------------------
  Line   lib/Doctrine/Common/Persistence/ObjectManagerDecorator.php
 ------ ---------------------------------------------------------------------------------------------------------------------------------
  50     Method Doctrine\Common\Persistence\ObjectManagerDecorator::persist() with return type void returns void but should not return
         anything.
  50     Result of method Doctrine\Common\Persistence\ObjectManager::persist() (void) is used.
  58     Method Doctrine\Common\Persistence\ObjectManagerDecorator::remove() with return type void returns void but should not return
         anything.
  58     Result of method Doctrine\Common\Persistence\ObjectManager::remove() (void) is used.
  74     Method Doctrine\Common\Persistence\ObjectManagerDecorator::clear() with return type void returns void but should not return
         anything.
  74     Result of method Doctrine\Common\Persistence\ObjectManager::clear() (void) is used.
  82     Method Doctrine\Common\Persistence\ObjectManagerDecorator::detach() with return type void returns void but should not return
         anything.
  82     Result of method Doctrine\Common\Persistence\ObjectManager::detach() (void) is used.
  90     Method Doctrine\Common\Persistence\ObjectManagerDecorator::refresh() with return type void returns void but should not return
         anything.
  90     Result of method Doctrine\Common\Persistence\ObjectManager::refresh() (void) is used.
  98     Method Doctrine\Common\Persistence\ObjectManagerDecorator::flush() with return type void returns void but should not return
         anything.
  98     Result of method Doctrine\Common\Persistence\ObjectManager::flush() (void) is used.
  130    Method Doctrine\Common\Persistence\ObjectManagerDecorator::initializeObject() with return type void returns void but should not
         return anything.
  130    Result of method Doctrine\Common\Persistence\ObjectManager::initializeObject() (void) is used.
 ------ ---------------------------------------------------------------------------------------------------------------------------------
```